logzio-sender/src/main/java/io/logz/sender/LogzioSender.java

```diff
@@ -39,16 +35,15 @@
     private ScheduledExecutorService tasksExecutor;
     private final int gcPersistedQueueFilesIntervalSeconds;
     private final AtomicBoolean drainRunning = new AtomicBoolean(false);
-    private final HttpsRequestConfiguration httpsRequestConfiguration;
-    private final HttpsSyncRequest httpsSyncRequest;
+    private final HttpsSyncSender httpsSyncSender;
 
     private LogzioSender(String logzioToken, String logzioType, int drainTimeout, int fsPercentThreshold, File bufferDir,
                          String logzioUrl, int socketTimeout, int connectTimeout, boolean debug,
                          SenderStatusReporter reporter, ScheduledExecutorService tasksExecutor,
                          int gcPersistedQueueFilesIntervalSeconds, boolean compressRequests)
             throws  LogzioParameterErrorException {
 
-        httpsRequestConfiguration = HttpsRequestConfiguration
+        HttpsRequestConfiguration httpsRequestConfiguration = HttpsRequestConfiguration
```

Comment:
The build() should be the last step. Usually it's xxBuilder.newBuilder().setY().setZ().build() --> xx